basic/src/test/java/org/jboss/hal/testsuite/test/runtime/batch/BatchManagementTestCase.java

```diff
@@ -0,0 +1,269 @@
+        deployBatchApplication(1000);
+        long executionId = batchOps.startJob(deploymentName, jobOneFileName);
+        batchOps.stopJob(deploymentName, executionId);
+        page.navigate2jobs();
+        ResourceTableRowFragment jobRow = page.selectRowForJob(jobOneFileName);
+        assertEquals(STOPPED, page.getBatchStatusFromRow(jobRow));
+
+        page.clickButton("Restart");
+        assertEquals(STARTED, page.getBatchStatusFromRow(page.getRowByExecutionId(String.valueOf(executionId + 1))));
+        String actualStatus = batchOps.getExecution(deploymentName, jobOneName, executionId + 1).batchStatus;
+        assertEquals(STARTED, actualStatus);
+    }
+
+    private void assertExecutionsMatchRows(Set<Execution> executionSet, Set<ResourceTableRowFragment> rowSet) {
+        assertEquals(executionSet.size(), rowSet.size());
+        executionSet.forEach(execution -> {
+            ResourceTableRowFragment actualRow = rowSet.stream().filter(row -> {
+                        return Long.valueOf(page.getExecutionIdFromRow(row)).equals(execution.executionId);
+                     }).findFirst().get();
+            assertEquals(execution.instanceId, (long) Long.valueOf(page.getInstanceIdFromRow(actualRow)));
+            assertEquals(execution.batchStatus, page.getBatchStatusFromRow(actualRow));
+            assertEquals(execution.startTime, page.getStartTimeFromRow(actualRow));
+        });
+    }
+
+    private void assertExecutionMatchJobDetail(Execution execution) {
+        MetricsAreaFragment jobStatusArea = page.getMetricsArea("Job Status");
+        assertAttributeEquals(String.valueOf(execution.instanceId), jobStatusArea.getMetric("Instance Id"));
+        assertAttributeEquals(execution.batchStatus, jobStatusArea.getMetric("Batch Status"));
+        assertAttributeEquals(execution.exitStatus, jobStatusArea.getMetric("Exit Status"));
+        assertAttributeEquals(execution.createTime, jobStatusArea.getMetric("Create Time"));
+        assertAttributeEquals(execution.startTime, jobStatusArea.getMetric("Start Time"));
+        assertAttributeEquals(execution.endTime, jobStatusArea.getMetric("End Time"));
+        assertAttributeEquals(execution.lastUpdatedTime, jobStatusArea.getMetric("Last Updated Time"));
+    }
+
+    private void assertAttributeEquals(String expectedAttrValueFromModel, String actualAttrValueFromBrowser) {
+        if (expectedAttrValueFromModel.equals("undefined")) {
+            assertEquals("", actualAttrValueFromBrowser);
+        } else {
+            assertEquals(expectedAttrValueFromModel, actualAttrValueFromBrowser);
+        }
+    }
+
+    private boolean everyJobFileNameInTableEquals(String expectedJobFileName) {
+        return page.getAllRows().stream().map(row -> page.getJobFileNameFromRow(row))
+                .allMatch(jobFileName -> jobFileName.equals(expectedJobFileName));
+    }
+
+    private void deployBatchApplication(long timeout) throws CommandFailedException {
```

Comment:
As heavily used method in tests, please provide javadoc. It is good to explain, how many batch jobs it contains.